src/main/com/yetanalytics/datasim/model.clj

```diff
@@ -0,0 +1,114 @@
+  (s/keys :req-un [::obj-override/objects]
+          :opt-un [::obj-override/weights]))
+
+(s/def ::model
+  (s/keys :opt-un [::alignments
+                   ::object-overrides]))
+
+(s/def ::default-model (s/nilable ::model))
+(s/def ::agent-models (s/map-of ::actor/actor-ifi ::model))
+(s/def ::group-models (s/map-of ::actor/actor-ifi ::model))
+(s/def ::role-models (s/map-of string? ::model))
+
+(def model-map-spec
+  (s/keys :req-un [::default-model
+                   ::agent-models
+                   ::group-models
+                   ::role-models]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Functions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; TODO: Temporal properties
+(defn- mapify-alignments
+  [alignments]
+  {:weights (reduce (fn [m {:keys [id weight]}] (assoc m id weight))
+                    {}
+                    alignments)})
+
+(defn- mapify-object-overrides
+  [object-overrides]
+  {:weights (reduce (fn [m {:keys [weight object]}] (assoc m object weight))
+                    {}
+                    object-overrides)
+   :objects (map :object object-overrides)})
+
+(s/fdef models->map
+  :args (s/cat :models ::model/models)
+  :ret model-map-spec)
+
+(defn models->map
+  [models]
+  (let [init-map       {:default-model nil
+                        :agent-models  {}
+                        :group-models  {}
+                        :role-models   {}}
+        persona-type-m {"Agent" :agent-models
+                        "Group" :group-models
+                        "Role"  :role-models}]
+    (reduce
+     (fn [acc {:keys [personae alignments objectOverrides]}]
+       (let [model* (cond-> {}
+                      (not-empty alignments)
+                      (assoc :alignments
+                             (mapify-alignments alignments))
+                      (not-empty objectOverrides)
+                      (assoc :object-overrides
+                             (mapify-object-overrides objectOverrides)))]
+         (if (some? personae)
+           (reduce
+            (fn [acc* {persona-id   :id
+                       persona-type :type}]
+              (let [persona-kw (get persona-type-m persona-type)]
+                (assoc-in acc* [persona-kw persona-id] model*)))
+            acc
+            personae)
+           (assoc acc :default-model model*))))
+     init-map
+     models)))
+
+(s/fdef get-actor-model
+  :args (s/cat :model-map model-map-spec
+               :agent-id ::actor/actor-ifi
+               :group-id ::actor/actor-ifi
+               :role-id  (s/and string? not-empty))
+  :ret ::model)
+
+(defn get-actor-model
+  [{:keys [default-model agent-models group-models role-models]}
+   agent-id
+   group-id
+   role-id]
+  ;; TODO: Figure out personae precedence
+  (or (get agent-models agent-id)
+      (get group-models group-id)
+      (get role-models role-id)
+      default-model))
```

Comment:
This is something I want to discuss: between an agent's Agent ID, Group ID, and Role, which one takes precedence if these personae are applied to different models?

Reply:
It seems like it should go from most-to-least specific to one person, so I'd think as you have it is good, but there may be some reason not to do that?

Reply:
That would make sense, but is "group" or "role" more specific?

Reply:
My hunch is that group is more specific, but it isn't strong one way or the other